ui/components/AutomationDetail.tsx

```diff
@@ -0,0 +1,117 @@
+import SubRouterTabs, { RouterTab } from "./SubRouterTabs";
+import SyncButton from "./SyncButton";
+
+type Props = {
+  automation?: Automation;
+  className?: string;
+  info: InfoField[];
+};
+
+const Info = styled.div`
+  margin-bottom: 16px;
+`;
+
+const TabContent = styled(Flex)`
+  margin-top: 52px;
+  width: 100%;
+  height: 100%;
+`;
+
+function AutomationDetail({ automation, className, info }: Props) {
+  const { notifySuccess } = React.useContext(AppContext);
+  const { path } = useRouteMatch();
+
+  const sync = useSyncAutomation({
+    name: automation?.name,
+    namespace: automation?.namespace,
+    clusterName: automation?.clusterName,
+    kind: AutomationKind.KustomizationAutomation,
+  });
+
+  const handleSyncClicked = (opts) => {
+    sync.mutateAsync(opts).then(() => {
+      notifySuccess("Resource synced successfully");
+    });
+  };
+
+  return (
+    <Flex wide tall column align className={className}>
+      <Flex wide between>
+        <Info>
+          <Heading level={2}>{automation?.namespace}</Heading>
+          <InfoList items={info} />
+        </Info>
+        <PageStatus
+          conditions={automation?.conditions}
+          suspended={automation?.suspended}
+        />
+      </Flex>
+      <Flex wide>
+        {sync.isError && (
+          <Alert
+            severity="error"
+            message={sync.error.message}
+            title="Sync Error"
+          />
+        )}
+      </Flex>
+      <SyncButton onClick={handleSyncClicked} loading={sync.isLoading} />
+      <TabContent>
+        <SubRouterTabs rootPath={`${path}/details`}>
+          <RouterTab name="Details" path={`${path}/details`}>
+            <ReconciledObjectsTable
+              automationKind={AutomationKind.KustomizationAutomation}
+              automationName={automation?.name}
+              namespace={automation?.namespace}
+              kinds={automation?.inventory}
+              clusterName={automation?.clusterName}
+            />
+          </RouterTab>
+          <RouterTab name="Events" path={`${path}/events`}>
+            <EventsTable
+              namespace={automation?.namespace}
+              involvedObject={{
+                kind: "Kustomization",
+                name: automation?.name,
+                namespace: automation?.namespace,
+              }}
+            />
+          </RouterTab>
+          <RouterTab name="Graph" path={`${path}/graph`}>
+            <ReconciliationGraph
+              automationKind={AutomationKind.KustomizationAutomation}
+              automationName={automation?.name}
+              kinds={automation?.inventory}
+              parentObject={automation}
+              clusterName={automation?.clusterName}
+            />
+          </RouterTab>
+        </SubRouterTabs>
+      </TabContent>
+    </Flex>
+  );
+}
+
+export default styled(AutomationDetail).attrs({
+  className: AutomationDetail.name,
+})`
+  width: 100%;
+
+  ${Alert} {
+    margin-bottom: 16px;
+  }
+`;
```

Comment:
Love this thanks for unifying HelmRelease and Kustomization 🌞

Reply:
Nice catch. I had turned off my `eslint` to save cycles. Not sure how that passed CI